src/Workspaces/Core/Portable/Serialization/SolutionReplicationContext.cs

```diff
@@ -1,31 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using Microsoft.CodeAnalysis.PooledObjects;
-using Roslyn.Utilities;
-
-namespace Microsoft.CodeAnalysis.Serialization;
-
-internal readonly struct SolutionReplicationContext : IDisposable
-{
-    private static readonly ObjectPool<ConcurrentSet<IDisposable>> s_pool = new(() => []);
-
-    private readonly ConcurrentSet<IDisposable> _resources;
-
-    public SolutionReplicationContext()
-        => _resources = s_pool.Allocate();
-
-    public void AddResource(IDisposable resource)
-        => _resources.Add(resource);
-
-    public void Dispose()
-    {
-        // TODO: https://github.com/dotnet/roslyn/issues/49973
-        // Currently we don't dispose resources, only keep them alive.
-        // Shouldn't we dispose them? 
-        // _resources.All(resource => resource.Dispose());
-        s_pool.ClearAndFree(_resources);
```

Comment:
note how we never dispose tehse objects.  *Nor should we*.   When serializing out to oop, just because something was written out to a MMF does not mean we should dispose that mmf, it's an entirely orthogonal concern outside of the scope of a remote call to oop.

The only value this type produced was to root those disposables, presumably to ensure that anything they were holding onto didn't go away over the lifetime of hte call.  However, none of that is necessary.  Either we're literally serializing out a value *by-value* over the wire.  In which case we don't need to keep it alive.  Or we're serializing it out by-reference (e.g. serializing out it's location in an MMF).  But in *all* those cases, the mmfs are never released anyways, so there's no need to keep anything alive.